src/BatteryStateOfChargeService.h

```diff
@@ -18,5 +18,22 @@ class BatteryStateOfChargeService : public I_BatteryStateOfChargeService
     // Convention is to have an _ after private members
     const double initialStateOfChargePercent_;
 
-    // Recommend adding necessary variables here
+    double InitialCurrent;
+    double CurrentPrime = 0;
```

Comment:
While this is valid c++11 code generally all initialization is done within the constructor. Think of the constructor as the setup() function within Processing 3. It is the best spot to initialize class variables within c++. 

BatteryStateOfChargeService::BatteryStateOfChargeService(double initialStateOfChargePercent)
 : initialStateOfChargePercent_(initialStateOfChargePercent)
 , CurrentPrime (0)
{
...
}

is the syntax to use. 

BatteryStateOfChargeService::BatteryStateOfChargeService(double initialStateOfChargePercent)
 : initialStateOfChargePercent_(initialStateOfChargePercent)
 {
   CurrentPrime = 0;
   ...
  }
is also valid but the other way is superior because of performance reasons (if you are curious https://isocpp.org/wiki/faq/ctors#init-lists)